### PR TITLE
Adding visual indicator of whether a question is required or optional

### LINKED ISSUE #1423: Add markers to optional and required questions in applicant view
In #1404 we describe how questions should get markers representing whether they're optional or required. Check that issue for the latest in UX plans, but at least for now:

* Put an asterisk next to required questions
* Put (optional) next to optional questions

This might be a lot, but we should err on the side of overcommunicating for now.

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
Showing whether or not a question is required by adding an asterisk or optional text at the end of the question.

### Description
Moved the creation of the question text div into it's own method. Adding either an asterisk or optional text based on how the question is configured. The optional text pulls from messages so it can be localized.

Sample of rendered questions:

image

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #1423

### CODE REVIEW

File: browser-test/src/program_deep_link.test.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe('navigating to a deep link', () => {
     // Assert
     await page.click('#continue-application-button')
     expect(await page.innerText('.cf-applicant-question-text')).toEqual(
-      questionText
+      questionText + "*"
```
Comment (shanemc-goog) [inline browser-test/src/program_deep_link.test.ts:53]: Note: I don't know if we handle RtL languages currently but if/when we do it'd be good to update this.

File: universal-application-tool-0.0.1/app/views/questiontypes/ApplicantQuestionRenderer.java [hunk 2]
```diff
@@ -85,4 +82,21 @@ Tag renderInternal(
         .with(questionTextDiv)
         .with(questionFormContent);
   }
+
+  private ContainerTag createQuestionTextTag(Messages messages) {
```
Comment (shanemc-goog) [inline universal-application-tool-0.0.1/app/views/questiontypes/ApplicantQuestionRenderer.java:86]: I don't see any new tests for showing the correct thing in the correct instance.

**Review by shanemc-goog — CHANGES_REQUESTED** [review]
Looks good, just need some tests.

File: universal-application-tool-0.0.1/conf/messages [hunk 3]
```diff
@@ -90,6 +90,9 @@ button.skipFileUpload=Skip
 # The text on the button an applicant clicks to upload a file and continue to the next screen.
 button.upload=Upload
 
+# The text shown next to a question indicating that it is optional
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply universal-application-tool-0.0.1/conf/messages:93]: She's working on translations for me already

**Comment by bion** [comment]
@gwendolyngoetz is it alright to put this PR back in draft mode?

**Pull request closed without merging**